server/src/main/java/org/elasticsearch/action/admin/indices/get/TransportGetIndexAction.java

```diff
@@ -109,14 +117,21 @@ protected void doMasterOperation(final GetIndexRequest request, String[] concret
             case SETTINGS:
                     if (!doneSettings) {
                         ImmutableOpenMap.Builder<String, Settings> settingsMapBuilder = ImmutableOpenMap.builder();
+                        ImmutableOpenMap.Builder<String, Settings> defaultSettingsMapBuilder = ImmutableOpenMap.builder();
                         for (String index : concreteIndices) {
                             Settings indexSettings = state.metaData().index(index).getSettings();
                             if (request.humanReadable()) {
                                 indexSettings = IndexMetaData.addHumanReadableSettings(indexSettings);
                             }
                             settingsMapBuilder.put(index, indexSettings);
+                            if (request.includeDefaults()) {
+                                Settings defaultIndexSettings =
+                                    settingsFilter.filter(indexScopedSettings.diff(indexSettings, Settings.EMPTY));
```

Comment:
To be honest I am not entirely sure about this either. I was following the example [here](https://github.com/elastic/elasticsearch/pull/29229/files#diff-e93d78a6550b215b579df596a20978c2R102).

Reply:
Ok. I'll play with it a little myself and see if I can prove to myself it doesn't matter. Or that it does.

Reply:
OK! That was pretty quick! I think this change is ok because of [this line](https://github.com/elastic/elasticsearch/blame/master/server/src/main/java/org/elasticsearch/common/settings/SettingsModule.java#L129).

Reply:
Thanks for the explanation!